web/index.html

```diff
@@ -1,378 +1,410 @@
-                acc++
-            }
-            return acc
-        }, 0)
-
-        const failedProviders = resp.length - successfulProviders
-
-        // Show providers with the data first, followed by reachable providers, then by those with addresses
-        resp.sort((a, b) => {
-            // First check if either has data available
-            const aHasData = a.DataAvailableOverBitswap?.Found || a.DataAvailableOverHTTP?.Found
-            const bHasData = b.DataAvailableOverBitswap?.Found || b.DataAvailableOverHTTP?.Found
-
-            if (aHasData && !bHasData) return -1
-            if (!aHasData && bHasData) return 1
-
-            // Then check connection errors
-            if (a.ConnectionError === '' && b.ConnectionError !== '') {
-                return -1;
-            } else if (a.ConnectionError !== '' && b.ConnectionError === '') {
-                return 1;
-            }
-
-            // Finally, show providers with addresses first
-            if(a.Addrs?.length > 0 && b.Addrs?.length === 0) {
-                return -1
-            } else if(a.Addrs?.length === 0 && b.Addrs?.length > 0) {
-                return 1
-            }
-
-            return 0
-        })
-
-        outText += `${successfulProviders > 0 ? '✅' : '❌'} Found ${successfulProviders} working providers (out of ${resp.length} provider records sampled from Amino DHT and IPNI) that could be connected to and had the CID available over Bitswap:`
-        for (const provider of resp) {
-            const couldConnect = provider.ConnectionError === ''
-            outText += `\n\t${provider.ID}\n`
-            if (provider.DataAvailableOverBitswap.Enabled === true) {
-	      outText += `\t\tLibp2p connected: ${couldConnect ? "✅" : `❌ ${provider.ConnectionError.replaceAll('\n', '\n\t\t')}` }`
-              outText += couldConnect ? `\n\t\tBitswap Check: ${provider.DataAvailableOverBitswap.Found ? `✅` : "❌"} ${provider.DataAvailableOverBitswap.Error || ''}` : ''
-            }
-            if (provider.DataAvailableOverHTTP.Enabled === true) {
-              const httpRes = provider.DataAvailableOverHTTP
-              outText += `\t\tHTTP Connected: ${httpRes.Connected ? '✅' : '❌'}`
-	      outText += `\n\t\tHTTP HEAD request: ${httpRes.Requested ? '✅' : '❌'}`
-	      outText += `\n\t\tHTTP Found: ${httpRes.Found ? '✅' : '❌'} ${provider.DataAvailableOverHTTP.Error || ''}`
-            }
-            outText += (couldConnect && provider.ConnectionMaddrs) ? `\n\t\tSuccessful Connection Multiaddr${provider.ConnectionMaddrs.length > 1 ? 's' : ''}:\n\t\t\t${provider.ConnectionMaddrs?.join('\n\t\t\t') || ''}` : ''
-            outText += (provider.Addrs.length > 0) ? `\n\t\tPeer Multiaddrs:\n\t\t\t${provider.Addrs.join('\n\t\t\t')}` : ''
-            outText += (typeof provider.Source === 'undefined') ? '' : `\n\t\tFound in: ${provider.Source}`
-        }
-
-        return outText
-    }
-</script>
-</body>
-</html>
+<!DOCTYPE html>
```

Comment:
The full file is diffed. Line-break change?

Could also take the chance to separate JS from HTML, even if we later template it into a single file.